### PR TITLE
🔧 migrate offers page to composition api

### PR DESCRIPTION
**Pull request by roiLeo**
- tested on `/bsx/offers`, `/snek/offers`
- migrate to vue composition api
- <s>⚠️ I am aware of whacked `AddressInput` </s> fixed with some neo magic

Screenshot 2023-07-12 at 09-23-58 KodaDot - NFT Market Explorer

### CODE REVIEW

File: components/bsx/Offer/MasterOfferTable.vue [hunk 1]
```diff
@@ -55,161 +54,130 @@
...
+  get: () => $route.query.tab || SelectedOfferType.ALL,
+  set: (value) => {
+    const { target } = $route.query
+    $router.replace({
+      query: { target, tab: value },
     })
+  },
+})
+
+const isOfferDropdownDisabled = computed(() => skipUserOffer)
```
Comment (Jarsen136) [inline components/bsx/Offer/MasterOfferTable.vue:88]:
```suggestion
const isOfferDropdownDisabled = computed(() => skipUserOffer.value)
```

**Review by Jarsen136 — APPROVED** [review]
code lgtm

File: components/bsx/Offer/MasterOfferTable.vue [hunk 2]
```diff
@@ -55,161 +54,130 @@
...
+    const { target } = $route.query
+    $router.replace({
+      query: { target, tab: value },
     })
+  },
+})
+
+const isOfferDropdownDisabled = computed(() => skipUserOffer.value)
+
+const OfferTypeOptions = ref([
```
Comment (vikiival) [inline components/bsx/Offer/MasterOfferTable.vue:90]: sir we know each other for a long time, 
but having variable with big letter makes me sad

File: components/bsx/Offer/MasterOfferTable.vue [hunk 3]
```diff
@@ -55,161 +54,130 @@
...
+    label: 'All Offers',
+  },
+  {
+    type: SelectedOfferType.CREATED,
+    label: 'Offers Created',
+  },
+  {
+    type: SelectedOfferType.INCOMING,
+    label: 'Offers Incoming',
+  },
```
Comment (vikiival) [inline components/bsx/Offer/MasterOfferTable.vue:102]: BTW make this simple string array and then traslate according to type?

**Review by vikiival — DISMISSED** [review]
smol

File: components/bsx/Offer/MasterOfferTable.vue (hunk 2 above)
Comment (roiLeo) [reply components/bsx/Offer/MasterOfferTable.vue:90]: Pascal Case ftw

**Review by preschian — APPROVED** [review]

**Comment by vikiival** [comment]
Sir please resolve congflicts and you have my approval to merge

Screenshot 2023-07-14 at 13 39 26

**Pull request merged**